### PR TITLE
6901 reduce named map calls

### PR DESCRIPTION
**Pull request by javitonino**
See #6901

Reduces number of named map saves during import 13->3 (two are actually needed, so just one extra).

I had to move map from the Relator to Member itself. Although Relator appears to memoize the relations, don't let it fool you, as a new Relator is created for each relation access. This appears to be a bug when extracting those methods to the Relator (2013) but a lot of code depends on this behaviour, specially visualization creation stuff.

Anyway, as I need map to be memoized (so `member.map.visualization` always refers to the same visualization instance) to be able to check if a named map update is already scheduled, I extracted it to Member. This is not a problem as Viz<->Map is 1:1 and never modified.

### CODE REVIEW

**Comment by xavijam** [comment]
good improvement! 👍

**Comment by javitonino** [comment]
Please CR @juanignaciosl. This is a bit dirty, but it is the best way I could think of doing this without some major changes to the old models. I added some explanation as to why some changes are needed in the description.

File: app/models/visualization/relator.rb [hunk 1]
```diff
@@ -8,20 +8,20 @@
...
+        others:           :other_layers,
+        named_map:        :named_maps_layers
+      }.freeze
+
+      INTERFACE = %w{ overlays user table related_templates related_tables related_canonical_visualizations
+                      layers stats mapviews total_mapviews single_data_layer? synchronization is_synced? permission
+                      parent children support_tables prev_list_item next_list_item likes likes_count reload_likes
+                      estimated_row_count actual_row_count }.freeze
+
+      def initialize(map, attributes = {})
```
Comment (juanignaciosl) [inline app/models/visualization/relator.rb]: Having `map` as a parameter and `map_id` inside `attributes` seems dangerous, both for edge cases (map not yet persisted, visualization not yet persisted and so on) and for possible mixed ids. Double check/review, please. Maybe some defensive code or logging makes sense here. Or better nothing, I can't tell, this just smells risky.

**Comment by juanignaciosl** [comment]
Well, this looks better than what you anticipated! Nevertheless there's a lot of class interface changes, and moving map from `Relator` to `Member` is potentially a huge change. I've only done a minor comment about having `map_id` decoupled from `map`, while both being parameters. LGTM, _but_ I'd run some smoke tests, specially those related to editor interactions, importing, and named/non named maps.

**Comment by javitonino** [comment]
Thanks! Probably was too deep in that well of negativity that is Member. The Relator -> Member change is scary (I initially tried to memoize the Relator and suddenly everything broke spectacularly), but seems to be working.

Sure on those smokes, at least create/rename/delete visualizations, as well as copy/import and changes in privacy seem most likely to break.

File: app/models/visualization/relator.rb (hunk 1 above)
Comment (javitonino) [inline app/models/visualization/relator.rb]: I think, I can remove `map_id` altogether and use only map; testing the change. I am pretty sure the map is always persisted before the visualization, so it should be alright, but I'll recheck some of the more exotic interactions such as copying.

Comment (juanignaciosl) [inline app/models/visualization/relator.rb]: Certainly it'd be the best fix.

**Pull request merged**